torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -1062,6 +1063,77 @@ Tensor* TensorExprKernel::computeValue(const torch::jit::Value* v) {
       });
     } break;
 
+    case aten::batch_norm: {
+      bool noWeight = false;
+      bool noBias = false;
+
+      if (v->node()->input(1)->node()->kind() == prim::Constant) {
+        const auto val = toIValue(v->node()->input(1)).value();
+        if (val.isNone()) {
+          noWeight = true;
+        }
+      }
+
+      if (v->node()->input(2)->node()->kind() == prim::Constant) {
+        const auto val = toIValue(v->node()->input(2)).value();
+        if (val.isNone()) {
+          noBias = true;
+        }
+      }
+
+      auto const& shape = valueShape(v->node()->inputs()[0]);
+      return Compute(
+          "aten_batch_norm",
+          c10::fmap<DimArg>(shape),
+          [this, v, noWeight, noBias](const std::vector<VarHandle>& axes) {
+            TORCH_INTERNAL_ASSERT(axes.size() >= 2);
+            auto const& n = v->node();
+
+            // parameter list: input, weight, bias, mean, var, training,
+            // momentum, eps, cudnn_enabled
+            Tensor* input = tensors_.at(n->inputs()[0]->unique());
+            Tensor* mean = tensors_.at(n->inputs()[3]->unique());
+            Tensor* var = tensors_.at(n->inputs()[4]->unique());
+            ExprHandle eps = constant(n->inputs()[7]);
+
+            // axes: N, C, H, W
+            std::vector<VarHandle> c(axes.begin() + 1, axes.begin() + 2);
+            if (noWeight && noBias) {
+              auto inv_var = rsqrt(var->call(c) + eps);
+              auto output =
+                  input->call(axes) * inv_var - mean->call(c) * inv_var;
+              return demoteOutput(output, n->output());
+            } else if (noWeight) {
+              Tensor* bias = tensors_.at(n->inputs()[2]->unique());
+
+              auto inv_var = rsqrt(var->call(c) + eps);
+              auto bias_v = bias->call(c);
+              auto beta = bias_v - mean->call(c) * inv_var;
+              auto output = input->call(axes) * inv_var + beta;
+              return demoteOutput(output, n->output());
+            } else if (noBias) {
+              Tensor* weight = tensors_.at(n->inputs()[1]->unique());
+
+              auto inv_var = rsqrt(var->call(c) + eps);
+              auto weight_v = weight->call(c);
+              auto alpha = inv_var * weight_v;
+              auto output = input->call(axes) * alpha - mean->call(c) * alpha;
+              return demoteOutput(output, n->output());
```

Comment:
I wonder if we need some promoteInputs as well.  I'm not certain here b/c pytorch type promotion is complicated, but I wonder what you get if, e.g., `input` is a `FloatTensor` and `mean` is a `DoubleTensor`.  (Not something that's likely to happen but we should handle it correctly all the same).